bulbs/content/tasks.py

```diff
@@ -51,3 +65,144 @@ def update_feature_type_rates(featuretype_pk):
+        })
+
+    if not post.ok:
+        logger.error('''
+            Error in posting Instant Article.\n
+            Content ID: {0}\n
+            IA ID: {1}\n
+            Status Code: {2}'''.format(
+                content.id,
+                content.instant_article_id,
+                post.status_code))
+        return
+
+    # Poll for status of article
+    response = ""
+    while response != "SUCCESS":
+        time.sleep(1)
+
+        status = requests.get('{0}/{1}?access_token={2}'.format(
+            fb_api_url,
+            post.json().get('id'),
+            fb_access_token
+        ))
+
+        # log errors
+        if not status.ok or status.json().get('status') == "ERROR":
+            logger.error('''
+                Error in getting status of Instant Article.\n
+                Content ID: {0}\n
+                IA ID: {1}\n
+                Status Code: {2}'''.format(
+                    content.id,
+                    content.instant_article_id,
+                    status.status_code))
+            return
+
+        response = status.json().get('status')
+
+    # set instant_article_id to response id
+    Content.objects.filter(pk=content.id).update(
+        instant_article_id=status.json().get('id'))
+
+
+def delete_article(content, fb_api_url, fb_access_token):
+    delete = requests.delete('{0}/{1}?access_token={2}'.format(
+        fb_api_url,
+        content.instant_article_id,
+        fb_access_token
+    ))
+
+    if not delete.ok:
+        logger.error('''
+            Error in deleting Instant Article.\n
+            Content ID: {0}\n
+            IA ID: {1}\n
+            Status Code: {2}'''.format(
+                content.id,
+                content.instant_article_id,
+                delete.status_code))
+    else:
+        status = delete.json().get('success')
+        if bool(status) is not True:
+            logger.error('''
+                Error in deleting Instant Article.\n
+                Content ID: {0}\n
+                IA ID: {1}\n
+                Error: {2}'''.format(
+                    content.id,
+                    content.instant_article_id,
+                    delete.json()))
+
+
+@shared_task(default_retry_delay=5, time_limit=300)
+def post_to_instant_articles_api(content_pk):
+    from .models import Content
+    content = Content.objects.get(pk=content_pk)
+
+    fb_page_id = getattr(settings, 'FACEBOOK_PAGE_ID', None)
+    fb_api_url = getattr(settings, 'FACEBOOK_API_BASE_URL', None)
+    fb_access_token = vault.read(settings.FACEBOOK_TOKEN_VAULT_PATH)
+
+    if not fb_page_id or not fb_api_url or not fb_access_token:
+        logger.error('''
+            Error in Django Settings.\n
+            FACEBOOK_PAGE_ID: {0}\n
+            FACEBOOK_API_BASE_URL: {1}\n
+            FB_ACCESS_TOKEN: {2}'''.format(
+                fb_page_id,
+                fb_api_url,
+                fb_access_token))
+        return
+
+    # render page source
+    context = {
```

Comment:
ok, i just moved the ordering around so that it always renders when it should, but only posts when it's in prod